api/client/compose.go

```diff
@@ -152,6 +156,35 @@ func (cli *DockerCli) CmdComposeUp(args ...string) error {
 	if *projectName == "" {
 		*projectName = getBaseDir()
 	}
+	for _, svconfig := range c.M {
+		if svconfig.Fip == ComposeFipAuto {
+			ctx := context.Background()
+			fipFilterArgs, _ := filters.FromParam("dangling=true")
```

Comment:
If a user run 2 parallel `compose up` with `fip:auto`, then it may introduce a collision. If this is an expected behavior, we should update doc for it. Otherwise, we should move the implementation to server side.

And if there are more than one `auto` FIPs, it's better to get them together rather than call the list API for each.